scripts/go-mod-tidy.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+go mod tidy
+if ! git diff --exit-code go.mod go.sum ; then
+    echo -e "\nPlease run 'go mod tidy' to clean up dependencies"
```

Comment:
This script is already running `go mod tidy`, so is there a reason why it should be run again as echoed in L5?

Reply:
This script is also used by the CI so this is the message to let users know how to pass this check.